torq.q

```diff
@@ -460,11 +460,16 @@ if[not any `debug`noredirect in key params; rolllogauto[]];
 
 // utilities to load individual files / paths, and also a complete directory
 // this should then be enough to bootstrap
-loadf:{
-	.lg.o[`fileload;"loading ",x];
+loadedf:enlist enlist""
+loadf0:{[reload;x]
+    if[not[reload]&x in loadedf;.lg.o[`fileload;"already loaded ",x];:()];
+    loadedf,:enlist x;
+    .lg.o[`fileload;"loading ",x];
 	
-	$[`debug in key .proc.params; system"l ",x;@[system;"l ",x;{.lg.e[`fileload;"failed to load",x," : ",y]}[x]]];
-	.lg.o[`fileload;"successfully loaded ",x]}
+    $[`debug in key .proc.params; system"l ",x;@[system;"l ",x;{.lg.e[`fileload;"failed to load",x," : ",y]}[x]]];
+    .lg.o[`fileload;"successfully loaded ",x]}
```

Comment:
If debugging is off and the above line fails then it looks like this will log as failed then as successful if I'm reading it right. Perhaps an explicit return is required?

Reply:
hmm, yes I think a bit of a refactor is required here